src/transaction/verify_version.rs

```diff
@@ -15,9 +15,9 @@ pub fn verify_version(
     }
 
     if *version == 0.into() || *version == *QUERY_VERSION_BASE {
-        if max_fee != 0 {
-            return Err(TransactionError::InvalidMaxFee);
-        }
+        // if max_fee != 0 {
+        //     return Err(TransactionError::InvalidMaxFee);
+        // }
```

Comment:
We need to find a proper fix for this before merging.

Reply:
I'll be honest, I have no idea how a transaction which didn't pass this check made it into the blockchain. But if there was nothing wrong with the transaction maybe we missed something about this check in particular?

Reply:
I think I found the problem. It's not that it made it to the chain with non-zero fee, but rather that we use `u128::MAX` as a flag to indicate the max fee should be ignored, only to later forget of this special case.
See the function `Declare::create_for_simulation`.

Reply:
There's a fix for this in main. It should work without commenting.